virtcontainers/hyperstart_agent.go

```diff
@@ -329,6 +329,20 @@ func (h *hyper) createSandbox(sandbox *Sandbox) (err error) {
 	return h.configure(sandbox.hypervisor, "", h.getSharePath(sandbox.id), false, nil)
 }
 
+func (h *hyper) cleanupSandbox(sandbox *Sandbox) {
+	if sandbox == nil || sandbox.id == "" {
+		h.Logger().Warn("sandbox is nil so exit on cleanup sandbox")
+		return
+	}
+
+	sharedPath := h.getSharePath(sandbox.id)
```

Comment:
do we have a chance that the shared dir is still mounted?

Reply:
wow, it seems we need an `EnsureRemoveAll()` to wrap `os.Removall()` to handle such scenario :face_with_head_bandage:

Reply:
related: #912